Integrate OAuth authentication custom claim check (#4072)
strimzi/strimzi-kafka-operator: documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc

<7> Set this to `false` in order to disable the SASL_OAUTHBEARER mechanism on the listener. At least one of SASL_PLAIN or SASL_OAUTHBEARER has to be enabled. Default is `true`.
<8> Set this to `true` in order to enable the SASL_PLAIN mechanism on the listener, which is supported by all clients on all platforms. The Kafka client has to enable the SASL_PLAIN mechanism and set the `username` and the `password`. This mechanism can be used to authenticate either by using the OAuth access token, or by using the OAuth clientId and secret (client credentials). If the client sets `username` to `access-token` then the password is interpreted as access token on the server, otherwise the user is interpreted as the clientId, and password as the client secret. Default is `false`.
<9> This has to be set to support the client credentials authentication when `enablePlain` is set to true, as described in previous point.

> <10> Additional custom rules can be imposed on the JWT access token during validation by setting this to a JsonPath filter query. If matching fails, the access token is rejected. When using `introspectionEndpointUri` the custom check is applied to the introspection endpoint response JSON.

Reviewer: ```suggestion
<10> Additional custom rules can be imposed on the JWT access token during validation by setting this to a JsonPath filter query. If the access token does not contain the necessary data, it is rejected. When using the `introspectionEndpointUri`, the custom check is applied to the introspection endpoint response JSON.
```
